app/src/main/cpp/Widget.cpp

```diff
@@ -70,7 +70,7 @@ struct Widget::State {
   void Initialize(const int aHandle, const WidgetPlacementPtr& aPlacement, const int32_t aTextureWidth, const int32_t aTextureHeight,
                   const QuadPtr& aQuad, const CylinderPtr& aCylinder) {
     handle = (uint32_t)aHandle;
-    name = "crow::Widget-" + std::to_string(handle);
+    name = aPlacement->name + '_' + std::to_string(handle);
```

Comment:
This is a good change but maybe we need to check that `name` is not null.

As far as I can see, if the value is null in Java then it will uninitialized in C++ (see `GET_STRING_FIELD` in `WidgetPlacement.cpp`).

Reply:
hmm in the practice this is actually enforced. Actually name is a std::string so is isn't that it is uninitialized but just an empty string.